### PR TITLE
Compatibility with GHC 8.6.2

### PR DESCRIPTION
**Pull request by pepeiborra**
## Summary

Build with GHC 8.6.2 and Stackage nightly

### CODE REVIEW

File: dejafu-tests/lib/Examples/ParMonad/Direct.hs [hunk 1]
```diff
@@ -268,31 +268,34 @@ runParIO userComp = do
...
+                              writeIORef topSessFlag True
+                              putMVar mfin x
+
+                    unregisterWorker tid
+                  ------------------------------------------------------------END WORKER THREAD
+                  pure (if cpu == main_cpu then Nothing else Just workerDone)
+
+            takeMVar mfin -- Final value.
+
+         other -> error "sessions"
```
Comment (barrucadu) [inline dejafu-tests/lib/Examples/ParMonad/Direct.hs]: Could you use `_` here, to fix the unused binding warning?

File: dejafu-tests/lib/Examples/ParMonad/Direct.hs [hunk 2]
```diff
@@ -445,14 +448,17 @@ rescheduleR cnt kont = do
...
+                  case sessions of
+                    Session _ finRef:_ -> do
+                      fin <- lift $ readIORef finRef
+                      if fin
+                      then kont (error "Direct.hs: The result value from rescheduleR should not be used.")
+                      else do
+                        lift $ steal mysched
+                        lift   yield
+                        rescheduleR (cnt+1) kont
+                    sessions -> error "sessions"
```
Comment (barrucadu) [inline dejafu-tests/lib/Examples/ParMonad/Direct.hs]: Similarly here

Comment (pepeiborra) [reply dejafu-tests/lib/Examples/ParMonad/Direct.hs]: Sure thing. 

Any plans for a Hackage release ?

Comment (barrucadu) [reply dejafu-tests/lib/Examples/ParMonad/Direct.hs]: Yes, I'll put one out today.

**Pull request merged**